ambry-api/src/main/java/com.github.ambry/config/PerformanceConfig.java

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright 2019 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.config;
+
+import com.github.ambry.commons.PerformanceIndex;
+import com.github.ambry.commons.Threshold;
+import com.github.ambry.rest.RestMethod;
+import com.github.ambry.utils.Pair;
+import java.util.EnumMap;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.Map;
+import org.json.JSONObject;
+
+
+/**
+ * Configuration parameters required to help evaluate performance of Ambry(i.e latency, health).
+ */
+public class PerformanceConfig {
+
+  public static final String NON_SUCCESS_REST_REQUEST_TOTAL_TIME_THRESHOLD_STR =
+      "performance.non.success.rest.request.total.time.threshold";
+  public static final String SUCCESS_REST_REQUEST_TOTAL_TIME_THRESHOLD_STR =
+      "performance.success.rest.request.total.time.threshold";
+  private static final String DEFAULT_REST_REQUEST_TOTAL_TIME_THRESHOLD_STR =
+      "{\"PUT\": \"Long.MAX_VALUE\",\"GET\": \"Long.MAX_VALUE\",\"POST\": \"Long.MAX_VALUE\",\"HEAD\": \"Long.MAX_VALUE\",\"DELETE\": \"Long.MAX_VALUE\"}";
+
+  /**
+   * The round trip time (millisecond) threshold for each type of rest request with non-success response(3xx or 4xx status code).
+   * The request is considered "Unsatisfied" if total time exceeds this threshold.
+   */
+  @Config(NON_SUCCESS_REST_REQUEST_TOTAL_TIME_THRESHOLD_STR)
+  @Default(DEFAULT_REST_REQUEST_TOTAL_TIME_THRESHOLD_STR)
+  private final String nonSuccessRestRequestTotalTimeThresholds;
```

Comment:
will we have separate thresholds for encrypted blob operations in the future?

What about things like getBlobInfo vs getBlob vs getSignedUrl. Or putBlob vs stitchBlob. I suppose we can start by setting such limits conservatively and then adding additional options in the future.

Reply:
This PR doesn't distinguish encrypted blob from non-encrypted one. The suggestion is good and I will raise a ticket to keep track of this. Maybe we can evaluate different types of requests in a more accurate way.